lib/events_processor.rb

```diff
@@ -17,8 +17,9 @@ def process
       Rails.logger.error errors unless errors.empty?
     end
 
-    def retrieve_reviews
-      Event.where('data ?| array[:keys]', keys: Event::TYPES).find_each.lazy.each do |event|
+    def retrieve_entity_records(types)
+      types = Event::TYPES if types == 'all'
```

Comment:
Perhaps we can do something like `types ||= Event::TYPES` and `def process(entity = nil)` so that we don't have to pass a string (`all`)